app/helpers/admin_flash_helper.rb

```diff
@@ -1,13 +1,12 @@
 module AdminFlashHelper
-  
+
   def disabled_flash_ids_from_cookies
-    disabled_flash_ids = []
-    cookies.each do |name,value| 
-      match = name.match(/admin_flash_(\d*)/)
-      if match 
-        disabled_flash_ids << match[1].to_i
-      end  
-    end
-    disabled_flash_ids
+    admin_flash_cookies.map { |cookie| cookie[1].to_i }
+  end
+
+  private
+
+  def admin_flash_cookies
+    cookies.map { |name, _| name.match(/admin_flash_(\d*)/) }.compact
```

Comment:
Your regex allows `'admin_flash_'`  since your pattern matches 0 or more numbers. I think we should go with `\d+`.
For example, try this:
```ruby
 a = {:nil=>"disabled", 'admin_flash_20'=>"disabled", :nothing=>"disabled", "admin_flash_"=>"disabled"}
 matches = a.map { |name, _| name.match(/admin_flash_(\d*)/)}.compact
=> [#<MatchData "admin_flash_20" 1:"20">, #<MatchData "admin_flash_" 1:"">]
c.map {|d| d[1].to_i}
=> [20, 0]
```
We're getting an additional element 0 because of the wrong match and `"".to_i` is 0.